### PR TITLE
[gramofononline] new extractor

### PR DESCRIPTION
**Pull request by user706**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

New extractor for https://gramofononline.hu

### CODE REVIEW

File: youtube_dl/extractor/gramofononline.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                'url': 'https://gramofononline.hu/go/noise_reduction/' + source + '.mp3',
+                'ext': 'mp3'
+            }]
+        }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        for line in webpage.split("\n"):
```
Comment (dstftw) [inline youtube_dl/extractor/gramofononline.py]: No need for that.

File: youtube_dl/extractor/gramofononline.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            m = re.search(r'var\s+trackList\s*=\s*(\[.*\]);?\s*', line)
+            if m:
+                break
+        lineobjs = json.loads(m.group(1))
+
+        if len(lineobjs) > 1:
```
Comment (dstftw) [inline youtube_dl/extractor/gramofononline.py]: Always return a playlist.

File: youtube_dl/extractor/gramofononline.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            'artist': 'Johanna Gadski, Otto Goritz, ismeretlen zenekar',
+            'ext': 'mp3'
+        },
+        'params': {
+            'skip_download': True,
+        }
+    }]
+
+    def _get_entry(self, obj):
+        id1 = obj.get("id")
```
Comment (dstftw) [inline youtube_dl/extractor/gramofononline.py]: Single quotes.

File: youtube_dl/extractor/gramofononline.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        source = obj.get("source")
+        name = obj.get("name")
```
Comment (dstftw) [inline youtube_dl/extractor/gramofononline.py]: Mandatory. Read coding conventions.

File: youtube_dl/extractor/gramofononline.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        # subname = obj.get("subname")
+        # paralelname = obj.get("paralelname")
+        # record = obj.get("record")
+        # long1 = obj.get("long")
+        # genre = obj.get("genre")
+        # author = obj.get("author")
+        # state = obj.get("state")
+        # matrica = obj.get("matrica")
+        # publisher = obj.get("publisher")
+        # img = obj.get("img")
```
Comment (dstftw) [inline youtube_dl/extractor/gramofononline.py]: Remove all garbage or add proper metadata.

File: youtube_dl/extractor/gramofononline.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+import re
+import json
+
+
+class GramofonOnlineIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?gramofononline\.hu(/(listen.php\?.*track=)?(?P<id>[0-9]+))?'
```
Comment (dstftw) [inline youtube_dl/extractor/gramofononline.py]: id must be captured correctly in all cases.

Comment (user706) [reply youtube_dl/extractor/gramofononline.py]: That's here is just to account for the autogenerated playlist which you get when visiting, https://gramofononline.hu (i.e. in that specific case there is no id).

Comment (user706) [reply youtube_dl/extractor/gramofononline.py]: But when using  
https://gramofononline.hu/1401835664/papageno-duett or  
https://gramofononline.hu/listen.php?autoplay=true&track=1401835664  
the id is properly captured

File: youtube_dl/extractor/gramofononline.py (hunk 1 above)
Comment (user706) [reply youtube_dl/extractor/gramofononline.py]: Now doing:
```python
        video_id = self._match_id(url)
	webpage = self._download_webpage(url, video_id)

        m = re.search(r'var\s+trackList\s*=\s*(\[.*\]);', webpage)
```

Is this what you meant?

File: youtube_dl/extractor/gramofononline.py (hunk 2 above)
Comment (user706) [reply youtube_dl/extractor/gramofononline.py]: ok

File: youtube_dl/extractor/gramofononline.py (hunk 3 above)
Comment (user706) [reply youtube_dl/extractor/gramofononline.py]: ok

File: youtube_dl/extractor/gramofononline.py (hunk 4 above)
Comment (user706) [reply youtube_dl/extractor/gramofononline.py]: could you please elaborate on this point?

File: youtube_dl/extractor/gramofononline.py (hunk 5 above)
Comment (user706) [reply youtube_dl/extractor/gramofononline.py]: ok

File: youtube_dl/extractor/gramofononline.py (hunk 1 above)
Comment (dstftw) [reply youtube_dl/extractor/gramofononline.py]: Don't capture empty list. `_search_regex`, `_parse_json`. Read coding conventions.

File: youtube_dl/extractor/gramofononline.py (hunk 4 above)
Comment (dstftw) [reply youtube_dl/extractor/gramofononline.py]: Again: read. It's all explained.

**Comment by user706** [comment]
This has been fixed for quite some time now. Could someone review please...

**Pull request closed without merging**